src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Serialization/ObjectReaders.TagHelperReader.cs

```diff
@@ -20,7 +20,18 @@ private record struct TagHelperReader(TagHelperDescriptorBuilder Builder)
             (nameof(TagHelperDescriptor.Metadata), ReadMetadata));
 
         private static void ReadDocumentation(JsonDataReader reader, ref TagHelperReader arg)
-            => arg.Builder.Documentation = Cached(reader.ReadString());
+        {
+            var documentationObject = ReadDocumentationObject(reader);
+
+            if (documentationObject is string text)
+            {
+                arg.Builder.SetDocumentation(Cached(text));
+            }
+            else
+            {
+                arg.Builder.SetDocumentation(documentationObject as DocumentationDescriptor);
```

Comment:
Do we ever expect this cast to fail? If not, please use a version that will fail if it's not a DocumentationDescriptor or null.

Reply:
No, this can't ever fail.